.NET client update

stefanb995 opened the pull request:
Adding two new parameters to Managed Instance (ProxyOverride and PublicDataEndpointEnabled)
Link to the PR with swagger change on which this code is generated: [link](https://github.com/Azure/azure-rest-api-specs/pull/5138)

<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
<!--
Please add an informative description that covers that changes made by the pull request.

If you are regenerating your SDK based off of a new swagger spec, please add the link to the corresponding swagger spec pull request that has been merged in the azure-rest-api-specs repository
-->

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] Please add REST spec PR link to the SDK PR
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

stefanb995 committed:
.NET client update

*Adding two new parameters to Managed Instance (ProxyOverride and PublicDataEndpointEnabled)

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Azure/azure-sdk-for-net?pullRequest=5197) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Azure/azure-sdk-for-net?pullRequest=5197) before we can accept your contribution.<br/><br/>:x: stefanb995 [sign now](https://cla.opensource.microsoft.com/Azure/azure-sdk-for-net?pullRequest=5197)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Azure/azure-sdk-for-net?pullRequest=5197) it.</sub>

stefanb995 committed:
Delete build.err

stefanb995 committed:
Delete build.wrn

dsgouda commented:
@jaredmoo please approve before I post my review.

jaredmoo reviewed (changes requested):
Discussed offline - needs enum for proxy property.

stefanb995 committed:
.NET client update

jaredmoo reviewed (changes requested):
Looks good so far, please update at least 1 scenario test to use the new properties

shahabhijeet committed:
Merge branch 'master' into AddProxyOverrideAndPublicEndpoint

stefanb995 committed:
Update ManagedInstanceCrudScenarioTests.cs

Updated test to include parameters ProxyOverride and PublicDataEndpointEnabled

stefanb995 committed:
Merge branch 'AddProxyOverrideAndPublicEndpoint' of https://github.com/stefanb995/azure-sdk-for-net into AddProxyOverrideAndPublicEndpoint

stefanb995 commented:
Updated existing scenario test adding these two new parameters.

dsgouda commented on src/SDKs/SqlManagement/Management.Sql/Generated/SqlManagementClient.cs:
```diff
@@ -565,33 +552,6 @@ public SqlManagementClient(ServiceClientCredentials credentials, params Delegati
-        /// <exception cref="System.ArgumentNullException">
-        /// Thrown when a required parameter is null
-        /// </exception>
-        public SqlManagementClient(ServiceClientCredentials credentials, HttpClient httpClient, bool disposeHttpClient) : this(httpClient, disposeHttpClient)
```
This is a breaking change, please bump version to `1.28.0-preview`

jaredmoo replied:
Deepak, is this changed expected from latest autorest? Just want to check.

jaredmoo replied:
Stefan, when you update csproj with new version, please also wipe out old release notes from 1.27.0 and add your new release notes.

dsgouda replied:
I'm not a 100% sure, but this is coming from AutoRest. @stefanb995 could you install latest autorest version and regen the code (`autorest --reset`)

dsgouda commented on src/SDKs/SDKs/_metadata/sql_resource-manager.txt:
```diff
@@ -3,12 +3,12 @@ AutoRest installed successfully.
 Commencing code generation
 Generating CSharp code
 Executing AutoRest command
-cmd.exe /c autorest.cmd https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/readme.md --csharp --version=latest --reflect-api-versions --csharp-sdks-folder=E:\git\azure-sdk-for-net\src\SDKs
```
There should be only 1 .txt file modified. Please checkout both .txt files and regenerate the code when the spec is merged.

stefanb995 committed:
Delete sql_resource-manager.txt

stefanb995 committed:
.NET sdk

SDK ganerated with option -AutoRestVersion "latest" on master azure-rest- api-specs

autorest version: 2.0.4283

stefanb995 committed:
Merge branch 'AddProxyOverrideAndPublicEndpoint' of https://github.com/stefanb995/azure-sdk-for-net into AddProxyOverrideAndPublicEndpoint

stefanb995 committed:
Update sql_resource-manager.txt

stefanb995 commented:
Hey guys, I have regenerated code from master branch. 
autorest version is 2.0.4283. 
There are some new files generated. Can you please take a look and tell me is this ok.

jaredmoo commented:
Please update csproj with new version (1.28.0-preview) and please also wipe out old release notes from 1.27.0 and add your new release notes.

jaredmoo commented:
Looks like you picked up sensitivity labels from @bashahee 's Swagger change ( https://github.com/Azure/azure-rest-api-specs/pull/5148 ) , but we can't add that to SDK without tests. @dsgouda , what is your guidance when features get tied up like this?

stefanb995 committed:
Update Microsoft.Azure.Management.Sql.csproj

stefanb995 commented:
Updated csproj.

Please let me know what could I do on my side. Should I just delete these new files and everything added for sensitivity labels?

dsgouda commented:
> 
> 
> Looks like you picked up sensitivity labels from @bashahee 's Swagger change ( [Azure/azure-rest-api-specs#5148](https://github.com/Azure/azure-rest-api-specs/pull/5148) ) , but we can't add that to SDK without tests. @dsgouda , what is your guidance when features get tied up like this?

@shahabhijeet could you please take a look

dsgouda commented:
@jaredmoo If I understand correctly, you are not ready to release the SesitivityLabels feature yet but you would like to release the changes in this PR.
The recommended way to do this is to remove the sensitivityLabel features from the default tag in readme.md so that generated code does not include that spec. Let us know if you have any doubts here

bashahee commented:
@jaredmoo @dsgouda Can we please merge these clients? I updated readme.md and swagger files in order to generate clients (our GA is at end of the month), but I see that @stefanb995 already did that, so can our clients be merged along with current PR?

jaredmoo commented:
@bashahee we need e2e scenario tests for all APIs, Can you please implement those?

bashahee commented:
> @bashahee we need e2e scenario tests for all APIs, Can you please implement those?

Yes, I will implement E2E scenario tests once this PR is merged. Is that what you meant?

jaredmoo commented:
I mean can you make your own PR that has your own scenario tests? We can merge them both together. (I can help resolve merge during Redmond hours if needed)

stefanb995 commented:
Guys, I need this merged as soon as possible. I need this change to be published to nuget before I send PR for powershell update.

dsgouda commented:
@stefanb995 @jaredmoo please ensure that the code was regenerated after @bashahee 's changes so that this PR has exactly what you plan to publish in the SDK?
Also please fix the failing tests, please check if the tests are running locally in the Playback mode

dsgouda commented:
@stefanb995 also link the Rest spec PR for which this SDK is being generated

stefanb995 committed:
Update of session record for updated test

stefanb995 committed:
Update TestCreateUpdateGetDropManagedInstance.json

stefanb995 commented:
@dsgouda I generated this SDK on master branch of Rest spec. I have also attached PR for my Rest spec change.

I forgot to commit session record for the test I updated. I have updated it now.  When I run tests in playback mode all tests are passing and 5 tests are being skipped.

dsgouda commented:
@azuresdkci Test azure-sdk-for-net.client please

dsgouda commented:
@azuresdkci retest this please

jaredmoo commented:
We need to make sure that @bashahee  has scenario tests for sensitivity labels before publishing

stefanb995 commented:
Can @bashahee update AssemblyInfo.cs file with AssemblyFileVersion (1.28.0.0)?

bashahee commented:
@stefanb995 @jaredmoo 
I opened a PR containing scenario tests for sensitivity labels clients.
https://github.com/Azure/azure-sdk-for-net/pull/5240

Moreover, I updated AssemblyInfo.cs.
